src/AutoMapper/TypePair.cs

```diff
@@ -185,6 +180,13 @@ public int Compare(Type x, Type y)
                 return 0;
             }
         }
+    }
 
+    public static class HashCodeCombiner
+    {
+        public static int Combine<T1, T2>(T1 obj1, T2 obj2) =>
+            CombineCodes(obj1.GetHashCode(), obj2.GetHashCode());
+
+        public static int CombineCodes(int h1, int h2) => ((h1 << 5) + h1) ^ h2;
```

Comment:
```
public static int Combine(object obj1, object obj2, params object[] addObjs) =>
            addObjs.Aggregate(obj1.GetHashCode().CombineCodes(obj2.GetHashCode()), (code, obj) => code.CombineCodes(obj.GetHashCode()));

public static int CombineCodes(this int h1, int h2) => ((h1 << 5) + h1) ^ h2;
```

Maybe make Combine extension as well and rename CombineHashCode.
Aggregate maybe too hard to read, just wanted to keep it one line.